File: spikes/issues/86/java/Probe.java
// Spike for issue #86 (Java): the issue's table omits Java entirely. Test it:
// is there an overload without a Toolkit, and does `null` survive the loop?
import io.github.muthuishere.toolnexus.*;

public class Probe {
    public static void main(String[] args) throws Exception {
        String base = System.getenv("SPIKE86_BASE");
        LlmClient.Options o = new LlmClient.Options();
        o.baseUrl = base; o.style = "openai"; o.model = "mock"; o.apiKey = "not-a-real-key";
        LlmClient client = LlmClient.create(o);

        // There is no run(String) overload, so `null` is the only spelling.
        try {
            LlmClient.RunResult r = client.run("write me a haiku", (Toolkit) null);
            System.out.println("java: null-toolkit run OK, text=" + r.text);
        } catch (Throwable e) {
            System.out.println("java: null-toolkit run FAILED: " + e.getClass().getSimpleName() + ": " + e.getMessage());
        }

        // The workaround that exists today.
        Toolkit tk = Toolkit.create(new Toolkit.Options().builtins(false));
        LlmClient.RunResult r2 = client.run("write me a haiku", tk);
        System.out.println("java: Toolkit builtins(false) OK, tools=" + tk.tools().size() + " text=" + r2.text);
    }
}

File: spikes/issues/86/probe.exs
# Spike for issue #86 (Elixir): the issue says run/ask need "an arity that omits
# the toolkit". Test both the missing arity and a nil toolkit.
base = System.get_env("SPIKE86_BASE")
client = Toolnexus.Client.create(base_url: base, style: "openai", model: "mock", api_key: "not-a-real-key")

# 1. The arity a completion user reaches for: run(client, prompt).
try do
  apply(Toolnexus.Client, :run, [client, "write me a haiku"])
  |> then(&IO.puts("ex: run/2 OK, text=#{inspect(&1.text)}"))
rescue
  e -> IO.puts("ex: run/2 FAILED: #{inspect(e.__struct__)}: #{Exception.message(e)}")
end

# 2. Explicit nil toolkit on the existing arity.
try do
  r = Toolnexus.Client.run(client, "write me a haiku", nil, [])
  IO.puts("ex: nil-toolkit run OK, text=#{inspect(r.text)}")
rescue
  e -> IO.puts("ex: nil-toolkit run FAILED: #{inspect(e.__struct__)}: #{Exception.message(e)}")
end

# 3. The workaround that exists today.
tk = Toolnexus.Toolkit.build(builtins: false)
r = Toolnexus.Client.run(client, "write me a haiku", tk, [])
IO.puts("ex: Toolkit.build(builtins: false) OK, tools=#{length(Toolnexus.Toolkit.tools(tk))} text=#{inspect(r.text)}")
